test/unittest/tests/test_session_config.cpp

```diff
@@ -0,0 +1,66 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "wt_internal.h"
+#include "wrappers/mock_session.h"
+
+/*
+ * Check a basic configuration that sets and clears a flag.
+ */
+#define FLAG_TEST(config_param, flag)                                                    \
```

Comment:
Catch2's [SECTION](https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md) is a great way of running a set of tests which share common setup & teardown code.

For example, FLAG_TEST could be replaced with something like:

`static void
test_config_flag(WT_SESSION_IMPL *session, const std::string& config_param, const int flag)
{
    std::string param_true = config_param + "=true";
    std::string param_false = config_param + "=false";
    REQUIRE(__ut_session_config_int(session, param_true.c_str()) == 0);
    REQUIRE(F_ISSET(session, flag));
    REQUIRE(__ut_session_config_int(session, param_false.c_str()) == 0);
    REQUIRE(!F_ISSET(session, flag));
    REQUIRE(__ut_session_config_int(session, param_true.c_str()) == 0);
    REQUIRE(__ut_session_config_int(session, "") == 0);
    REQUIRE(F_ISSET(session, flag));
}

TEST_CASE("Session config - test setting and clearing a flag")
{
    /* Build Mock session, this will automatically create a mock connection. */
    std::shared_ptr<MockSession> session_mock = MockSession::buildTestMockSession();
    WT_SESSION_IMPL *session = session_mock->getWtSessionImpl();

    SECTION("ignore_cache_size")
    {
        test_config_flag(session, "ignore_cache_size", WT_SESSION_IGNORE_CACHE_SIZE);
    }

    SECTION("cache_cursors")
    {
        test_config_flag(session, "cache_cursors", WT_SESSION_CACHE_CURSORS);
    }

    SECTION("debug.checkpoint_fail_before_turtle_update")
    {
        test_config_flag(session, "debug.checkpoint_fail_before_turtle_update", WT_SESSION_DEBUG_CHECKPOINT_FAIL_BEFORE_TURTLE_UPDATE);
    }

    SECTION("debug.release_evict_page")
    {
        test_config_flag(session, "debug.release_evict_page", WT_SESSION_DEBUG_RELEASE_EVICT);
    }
}`

Reply:
838a57d7b

Reply:
Thanks!